src/openms/source/ANALYSIS/ID/IDMapper.cpp

```diff
@@ -299,107 +284,185 @@ namespace OpenMS
     // for statistics
     Size id_matches_none(0), id_matches_single(0), id_matches_multiple(0);
 
-    // iterate over the peptide IDs
+    // build map from file to peptide id
+//    std::map<String, std::unordered_map<String, const PeptideIdentification*>> file2nativeid2pepid;
+    bool has_spectrum_references{false};
     for (Size i = 0; i < ids.size(); ++i)
     {
-      if (ids[i].getHits().empty()) continue;
+      const PeptideIdentification* pid = &ids[i];
+
+      // skip IDs without peptide annotations
+      if (pid->getHits().empty()) continue;
+      String file_origin = pid->getMetaValue("file_origin", "");
```

Comment:
I think this is wrong. IDMerger annotates id_merge_idx
This can be used to look up in the ProteinIDRun getFileOrigin (or whatever it is called)